README.md

```diff
@@ -81,7 +81,9 @@ require 'pry'
 
 # Install Puppet on all hosts
 hosts.each do |host|
-  on host, install_puppet
+  # Select the correct method below.
+  on host, install_puppet  # this is for Puppet 3
+  install_puppet_agent_on(host, :version => '1.5.2') # this is for Puppet 4
```

Comment:
`install_puppet` is itself [a shortcut to installing puppet on all hosts available](https://github.com/puppetlabs/beaker/blob/master/lib/beaker/dsl/install_utils/foss_utils.rb#L205-L208). This means that for running it on all hosts, you're installing puppet hosts*hosts number of times.

This should actually be just `install_puppet` outside of the hosts loop, or if wanted inside the loop, it should be done with this line:
```ruby
install_puppet_on(host, options)
```
Note that the options are the global options (accessible from the shim as the `options` method used here).